CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+
+We may use deprecation notices to advise users to move towards newer APIs.
+
+Older APIs will continue being honored, though.
+
+### The `pflag` promise of being a drop-in replacement of `flag`
+
+`pflag` is a drop-in replacement for `go`'s flag package, implementing
+POSIX/GNU-style --flags. That's it.
+
+`pflag` will follow the evolution of the standard library `flag`, and continue to maintain the promise of being
+a _drop-in replacement_ for `flag` (and then some of course, but at least that).
+
+### Versioning
+
+This package abides by `semver`.
+
+Hence:
+* `v1.0.6` : patch
+* `v1.1.0` : new feature
+
+As stated above, it is unlikely we ever issue a `v2.0.0` (breaking change).
+
+## Pull requests
+
+All contributions are much appreciated and welcome. There is no small contributions: additional documentation, examples, 
+tests, etc are all valuable additions, not just new features.
+
+### Linting
+
+This repository uses `github.com/golangci/golangci-lint/cmd/golangci-lint` as meta-linter, enforced by CI.
+The set of rules may evolve over time. `revive` and `govet` are must-have.
+
+> TODO: agree on the initial .golangci.yml setup, with a no-nonsense approach to linting. Code quality without nitpicking... :)
+> As a started, here is a sample configuration that I use on a daily basis for personal projects (no strong agenda on this):
+> [golangci-lint.yml](https://github.com/fredbi/gflag/blob/master/.golangci.yml).
+
+### Testing
+
+Pull requests should include significant unit tests.
+
+This repository uses `github.com/stretchr/testify` framework for unit tests.
+
+> TODO: to be discussed, what we mean by "significant", e.g. >70% test coverage or so.
+
+### Documentation
+
+Keeping a compelling `godoc` documentation is part of the "drop-in replacement" promise.
+
+Any newly exposed API should come with a clear, readable description of what it does.
+
+Testable example are alway welcome (although not required).
+
+Conversely, internals only need comments if a particular caveat or expectation requires wording besides just code.
+
+### Dependencies
+
+So far, `pflag` has managed to build without _any_ external dependency.
+
+We may indulge into _a few_ dependencies, especially test dependencies, or things like `golang.org/x/...`.
+
+We should not need much more than that, unless we are talking about extensions (e.g. new flag types). See below.
+
+### Commit etiquette
+
+Please keep your commit title and description concise and to the point.
+
+Feel free to `git rebase -i`, `git commit --amend` etc, so this repository keeps a nice and readable git history,
+without merge commits or fixup rescindments.
+
+### Contributor Licence Agreement
+
+Contributions are subject to a CLA DCO being signed off.
+
+> TODO: is this something that is still needed? Anyhow, check if the CLA terms are up to date.
+> The Linux Foundation has introduced [EasyCLA](https://docs.linuxfoundation.org/lfx/easycla).
+> I believe that `cla.assistant.io` is a bit old.
```

Comment:
I'm not a huge fan of this. I see often that folks will make a quick PR and then not follow up with the CLA (due to time or whatever) and then the PR just waits indefinitely.

Reply:
Thats fair.